### PR TITLE
Use external policies to self-manage credentials

### PR DESCRIPTION
**Pull request by dav3r**
#  #

## 🗣 Description ##

This PR adds support for creating non-admin users that may or may not be required to use multi-factor authentication (MFA).  It removes the inline policy (`SelfManagedCredsWithMFA`) and replaces it with one of the policies created in https://github.com/cisagov/cool-accounts/pull/79.
 

## 💭 Motivation and context ##

Previously, all non-admin users were required to use MFA.  However, certain non-admin users require programmatic access to IAM roles, so those users can now be given the appropriate IAM policy to allow that.

This PR is dependent on the changes in https://github.com/cisagov/cool-accounts/pull/79.

## 🧪 Testing ##

I applied these changes in Production and verified that each user was given the appropriate policy (either MFA-required or no MFA required).

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
* [x] All new and existing tests pass.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ policies to them.  We recommend creating your Users account via the
 | aws_region | The AWS region where the non-global resources are to be provisioned (e.g. "us-east-1"). | `string` | `us-east-1` | no |
 | non_self_admin_users | A list containing the usernames of non-admin users that are not allowed to administer their own accounts.  Example: [ "service-account1", "service-account2", "service-account3" ] | `list(string)` | `[]` | no |
 | tags | Tags to apply to all AWS resources created | `map(string)` | `{}` | no |
-| users | A list containing the usernames of each non-admin user.  Example: [ "firstname1.lastname1", "firstname2.lastname2", "firstname3.lastname3" ] | `list(string)` | n/a | yes |
+| users | A map whose keys are the usernames of each non-admin user and whose values are a map containing supported user attribtes.  The only currently-supported attribute is "require_mfa" (boolean).  Example: { "firstname1.lastname1" = { "require_mfa" = false }, "firstname2.lastname2" = { "require_mfa" = true }, "firstname3.lastname3" = { "require_mfa" = false } } | `map(map(string))` | n/a | yes |
```
Comment (jsf9k) [inline README.md]: There is a typo here:
```suggestion
| users | A map whose keys are the usernames of each non-admin user and whose values are a map containing supported user attributes.  The only currently-supported attribute is "require_mfa" (boolean).  Example: { "firstname1.lastname1" = { "require_mfa" = false }, "firstname2.lastname2" = { "require_mfa" = true }, "firstname3.lastname3" = { "require_mfa" = false } } | `map(map(string))` | n/a | yes |
```

File: variables.tf [hunk 2]
```diff
@@ -5,9 +5,8 @@
 # ------------------------------------------------------------------------------
 
 variable "users" {
-  type = list(string)
-  # Currently-defined roles: financial_audit, security_audit
-  description = "A list containing the usernames of each non-admin user.  Example: [ \"firstname1.lastname1\", \"firstname2.lastname2\", \"firstname3.lastname3\" ]"
+  type        = map(map(string))
+  description = "A map whose keys are the usernames of each non-admin user and whose values are a map containing supported user attribtes.  The only currently-supported attribute is \"require_mfa\" (boolean).  Example: { \"firstname1.lastname1\" = { \"require_mfa\" = false }, \"firstname2.lastname2\" = { \"require_mfa\" = true }, \"firstname3.lastname3\" = { \"require_mfa\" = false } }"
```
Comment (jsf9k) [inline variables.tf]: Same typo:
```suggestion
  description = "A map whose keys are the usernames of each non-admin user and whose values are a map containing supported user attributes.  The only currently-supported attribute is \"require_mfa\" (boolean).  Example: { \"firstname1.lastname1\" = { \"require_mfa\" = false }, \"firstname2.lastname2\" = { \"require_mfa\" = true }, \"firstname3.lastname3\" = { \"require_mfa\" = false } }"
```

**Review by jsf9k — APPROVED** [review]
Approved, but please fix the two typos I noticed.

"Toki is right, sorry if we got very good ears."  - Skwisgaar Skwisgelf

**Review by mcdonnnj — APPROVED** [review]
Makes sense to me. 👍

**Pull request merged**